src/screenshot.cpp

```diff
@@ -680,8 +680,18 @@ static const char *MakeScreenshotName(const char *default_fn, const char *ext, b
 		}
 	}
 
-	/* Add extension to screenshot file */
 	size_t len = strlen(_screenshot_name);
+
+	/* Handle user-specified filenames ending in %d with automatic numbering */
+	if (len >= 2) {
```

Comment:
There is a `StrEndsWith` function, which would probably make the code a bit clearer. So instead of the len and character compares, just use `if (StrEndsWith(_screenshot_name, "%d")) {`.

Reply:
Thanks for the tip, I've switched the logic to just use `StrEndsWith`